python/benchmarks/bi_benchmarks.py

```diff
@@ -0,0 +1,149 @@
+        for num in BIBenchmarks.params:
+            _df = self.df.copy(deep=True)
+            if (num > 1):
+                # lets create N times bigger DF
+                dfcum = self.df.copy(deep=True)
+                for i in range(1, (BIBenchmarks.params[-1])):
+                    dfcum = pd.concat([dfcum, self.df])            
+                _df = dfcum
+            self.lib.write(f"{self.symbol}{num}", _df)
+
+        print("If pandas query produces different dataframe than arctic one stop tests!")
+        print("This will mean query problem is there most likely")
+
+        print("Pre-check correctness for query_groupaby_city_count_all")
+        _df = self.df.copy(deep=True)
+        arctic_df = self.time_query_groupaby_city_count_all(BIBenchmarks.params[0])
+        _df = self.get_query_groupaby_city_count_all(_df)
+        arctic_df.sort_index(inplace=True)
+        self.assert_frame_equal(_df, arctic_df)
+
+        print("Pre-check correctness for query_groupaby_city_count_isin_filter")
+        _df = self.df.copy(deep=True)
+        arctic_df = self.time_query_groupaby_city_count_isin_filter(BIBenchmarks.params[0])
+        _df = self.get_query_groupaby_city_count_isin_filter(_df)
+        arctic_df.sort_index(inplace=True)
+        self.assert_frame_equal(_df, arctic_df)
+
+        print("Pre-check correctness for query_groupaby_city_count_filter_two_aggregations")
+        _df = self.df.copy(deep=True)
+        arctic_df = self.time_query_groupaby_city_count_filter_two_aggregations(BIBenchmarks.params[0])
+        _df = self.get_query_groupaby_city_count_filter_two_aggregations(_df)
+        arctic_df.sort_index(inplace=True)
+        self.assert_frame_equal(_df, arctic_df)
+
+        print("All pre-checks completed SUCCESSFULLY")
+
+        del self.ac
+    
+    def assert_frame_equal(self, pandas_df:pd.DataFrame, arctic_df:pd.DataFrame):
+        arctic_df.sort_index(inplace=True)
+        pd.testing.assert_frame_equal(pandas_df,
+                                      arctic_df, 
+                                      check_column_type=False, 
+                                      check_dtype=False)
+
+    def setup(self, num_rows):
+        self.ac = Arctic(f"lmdb://opensource_datasets_{self.lib_name}?map_size=20GB")
+        self.lib = self.ac.get_library(self.lib_name)
+
+    def teardown(self, num_rows):
+        del self.ac
+
+    def time_query_readall(self, times_bigger):
+        self.lib.read(f"{self.symbol}{times_bigger}")
+
+    def get_query_groupaby_city_count_all(self, q):
+        return q.groupby("City").agg({"Keyword": "count"})
+               
+    def time_query_groupaby_city_count_all(self, times_bigger) -> pd.DataFrame:
+        q = QueryBuilder()
+        q = self.get_query_groupaby_city_count_all( q)
+        df = self.lib.read(f"{self.symbol}{times_bigger}", query_builder=q)
+        return df.data
+
+    def get_query_groupaby_city_count_isin_filter(self, q):
```

Comment:
Typo: should be `groupby`
Applies to all the methods with the same typo

Reply:
done!